android/src/main/java/cl/json/social/InstagramShare.java

```diff
@@ -61,18 +57,32 @@ protected void openInstagramUrlScheme(String url) {
             super.openIntentChooser();
     }
 
-    protected void openInstagramIntentChooserForImage(String url, String chooserTitle) {
+    protected void openInstagramIntentChooser(String url, String chooserTitle, Boolean isImage) {
         Boolean shouldUseInternalStorage = ShareIntent.hasValidKey("useInternalStorage", options) && options.getBoolean("useInternalStorage");
-        ShareFile shareFile = new ShareFile(url, "image/jpeg", "image", shouldUseInternalStorage, this.reactContext);
+        ShareFile shareFile = isImage 
+            ? new ShareFile(url, "image/jpeg", "image", shouldUseInternalStorage, this.reactContext) 
+            : new ShareFile(url, "video/mp4", "video", shouldUseInternalStorage, this.reactContext);
         Uri uri = shareFile.getURI();
 
         Intent feedIntent = new Intent(Intent.ACTION_SEND);
-        feedIntent.setType("image/*");
+
+        if (isImage) {
+            feedIntent.setType("image/*");
+        } else {
+            feedIntent.setType("video/*");
+        }
+
         feedIntent.putExtra(Intent.EXTRA_STREAM, uri);
         feedIntent.setPackage(PACKAGE);
 
         Intent storiesIntent = new Intent("com.instagram.share.ADD_TO_STORY");
-        storiesIntent.setDataAndType(uri, "jpg");
+
+        if (isImage) {
+            storiesIntent.setDataAndType(uri, "jpg");
+        } else {
+            storiesIntent.setDataAndType(uri, "mp4");
```

Comment:
This only accepts `mp4` files?

Reply:
Hey @MateusAndrade I've updated the PR to allow also other type of files besides the `jpg` and `mp4` format

Reply:
Ah, nice! Sorry for the delay, nice work! 😄

Reply:
Thanks! :)